src/render.js

```diff
@@ -42,104 +42,74 @@ const renderToSketch = (
+    // assume if name is set on this nested page, the intent is to overwrite
+    // the name of the page it is getting rendered into
+    if (tree.props.name) {
+      container.setName(tree.props.name);
+    }
 
-    return renderToSketch(tree, page);
+    return tree.children.map(child => renderToSketch(child, container));
   }
+  return renderToSketch(tree, container);
+};
 
-  // Keep track of created pages
-  // Starts at `1` by default, because there is always one default page per document
-  let pageTotal = symbolPage ? 2 : 1;
-  // Keep track of existing and created pages to pass back to function caller
-  const pages = [];
-
-  pageData.forEach((data) => {
-    // Get Current Page
-    let page = getDocument(context).currentPage();
-
-    if (pageTotal > 1) {
-      // Create new page
-      page = getDocument(context).addBlankPage();
-    } else {
-      pageTotal += 1;
-    }
+export const renderDocument = (tree: TreeNode): Array<?SketchLayer> => {
+  resetDocument();
 
-    if (data.name) {
-      // Name new page
-      page.setName(data.name);
-    }
+  const doc = getDocument(context);
+  const initialPage = doc.currentPage();
+  const shouldRenderInitialPage = isSymbolsPage(initialPage);
 
-    if (data.children && data.children.length > 0) {
-      // Clear out page layers to prepare for re-render
-      resetLayer(page);
-      data.children.forEach((child) => {
-        renderToSketch(child, page);
-      });
-    }
+  return tree.children.map((child, i) => {
+    const page =
+      i === 0 && !shouldRenderInitialPage ? initialPage : doc.addBlankPage();
 
-    pages.push(page);
+    renderContainer(child, page);
+    return page;
   });
-
-  return pages;
 };
 
 export const render = (
   element: React$Element<any>,
   container?: ?SketchLayer
 ): ?SketchLayer | Array<?SketchLayer> => {
-  if (appVersionSupported()) {
-    try {
-      // Clear out document or layer to prepare for re-render
-      if (!container) {
-        resetDocument();
-      } else {
-        resetLayer(container);
-      }
-
-      // Build out sketch compatible tree representation
-      const tree = buildTree(element);
-
-      // Traverse tree to create pages and render their children.
-      return buildPages(tree, container);
-    } catch (err) {
-      const tree = buildTree(<RedBox error={err} />);
-      return renderToSketch(tree, getDocument(context).currentPage());
+  if (!appVersionSupported()) {
+    return null;
+  }
+
+  // The Symbols page holds a special meaning within Sketch / react-sketchapp
+  // and due to how `makeSymbol` works, we cannot render into it
+  if (container && isSymbolsPage(container)) {
+    throw Error('Cannot render into Symbols page');
+  }
+
+  const doc = getDocument(context);
+  const currentPage = doc.currentPage();
+  try {
+    const tree = buildTree(element);
+
+    if (tree.type === 'document' && container) {
+      throw Error('Cannot render `Document` component into a container');
```

Comment:
I don't think that's right. What if I pass a document as the container? I might not want to render into the current document